doc/scripts/docgen.py

```diff
@@ -99,3 +103,6 @@ def call_sphinx(builder, workdir):
 
     # To go back to the original current directory.
     os.chdir(currentdir)
+
+    # Reset THEANO_FLAGS
+    os.environ['THEANO_FLAGS'] = env_th_flags
```

Comment:
This fail if `THEANO_FLAGS` was not set, in which case `env_th_flags` is `None`.